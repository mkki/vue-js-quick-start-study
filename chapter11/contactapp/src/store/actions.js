import Constant from '../constant';
import ContactsAPI from '../api/ContactsAPI';

export default {
    [Constant.ADD_CONTACT_FORM]: (store) => {
        store.commit(Constant.ADD_CONTACT_FORM);
    },
    async [Constant.ADD_CONTACT](store) {
        var response = await ContactsAPI.addContact(store.state.contact);
        if (response.data.status == 'success') {
            store.dispatch(Constant.CANCEL_FORM);
            store.dispatch(Constant.FETCH_CONTACTS, { pageno: 1 });
        } else {
            console.log(`연락처 추가 실패: ${response.data}`)
        }
    },
    async [Constant.EDIT_CONTACT_FORM](store, payload) {
        var response = await ContactsAPI.fetchContactOne(payload.no);
        store.commit(Constant.EDIT_CONTACT_FORM, { contact: response.data });
    },
    async [Constant.UPDATE_CONTACT](store) {
        var response = await ContactsAPI.updateContact(store.state.contact);
        var currentPageNo = store.state.contactlist.pageno;
        if (response.data.status == 'success') {
            store.dispatch(Constant.CANCEL_FORM);
            store.dispatch(Constant.FETCH_CONTACTS, { pageno: currentPageNo });
        } else {
            console.log(`연락처 변경 실패: ${response.date}`);
        }
    },
    async [Constant.EDIT_PHOTO_FORM](store, payload) {
        var response = await Constant.fetchContactOne(payload.no);
        store.commit(Constant.EDIT_PHOTO_FORM, { contact: response.data });
    },
    async [Constant.UPDATE_PHOTO](store, payload) {
        var currentPageNo = store.state.contactlist.pageno;
        var response = await ContactsAPI.updatePhoto(payload.no, payload.file);
        store.dispatch(Constant.CANCEL_FORM);
        store.dispatch(Constant.FETCH_CONTACTS, { pageno: currentPageNo });
    },
    async [Constant.FETCH_CONTACTS](store, payload) {
        var pageno;
        if (typeof payload === 'undefined' || typeof payload.pageno === 'undefined') {
            pageno = 1;
        } else {
            pageno = payload.pageno;
        }
        var pagesize = store.state.contactlist.pagesize;

        var response = await ContactsAPI.fetchContacts(pageno, pagesize);
        store.commit(Constant.FETCH_CONTACTS, { contactlist: response.data });
    },
    [Constant.CANCEL_FORM]: (store, payload) => {
        store.commit(Constant.CANCEL_FORM);
    },
    async [Constant.DELETE_CONTACT](store, payload) {
        var currentPageNo = store.state.contactlist.pageno;
        var response = await ContactsAPI.deleteContact(payload.no);
        store.dispatch(Constant.FETCH_CONTACTS, { pageno: currentPageNo });
    },
    [Constant.CHANGE_MODE]: (store, payload) => {
        store.commit(Constant.CHANGE_MODE, { mode: payload.mode });
    }
}