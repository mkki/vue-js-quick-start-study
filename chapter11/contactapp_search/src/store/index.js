import Vue from 'vue';
import Vuex from 'vuex';
import Constant from '../Constant';
import SearchApi from '../api/SearchAPI';

Vue.use(Vuex);

const store = new Vuex.Store({
    state: {
        contacts: []
    },
    mutations: {
        [Constant.SEARCH_CONTACT]: (state, payload) => {
            state.contacts = payload.contacts;
        }
    },
    actions: {
        [Constant.SEARCH_CONTACT]: (store, payload) => {
            SearchApi.searchContact(payload.name)
                .then((response) => {
                    store.commit(Constant.SEARCH_CONTACT, { contacts: response.data })
                })
        }

        // asnyc/await syntax
        // async [Constant.SEARCH_CONTACT](store, payload) {
        //   var response = await SearchApi.searchContact(payload.name);
        //   store.commit(Constant.SEARCH_CONTACT, { contacts: response.data });
        // }

    }
})

export default store;