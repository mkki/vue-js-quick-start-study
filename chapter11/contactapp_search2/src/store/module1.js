import Constant from '../Constant';
import SearchApi from '../api/SearchAPI';

export default {
    state: {
        contacts: []
    },
    mutations: {
        [Constant.SEARCH_CONTACT]: (state, payload) => {
            state.contacts = payload.contacts;
        }
    },
    actions: {
        async [Constant.SEARCH_CONTACT](store, payload) {
            var response = await SearchApi.searchContact(payload.name);
            store.commit(Constant.SEARCH_CONTACT, { contacts: response.data });
            if (response.data.length > 0) {
                store.dispatch(Constant.ADD_KEYWORD, payload);
            }
        }
    }
}