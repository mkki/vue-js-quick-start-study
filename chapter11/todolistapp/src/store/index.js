import Vue from 'vue';
import Vuex from 'vuex';
import Constant from '../constant';

Vue.use(Vuex);

const store = new Vuex.Store({
    state: {
        todolist: [
            { todo: "영화", done: false },
            { todo: "야구", done: false },
            { todo: "공부", done: false },
            { todo: "카페", done: false }
        ]
    },
    mutations: {
        [Constant.ADD_TODO]: (state, payload) => {
            if (payload.todo !== "") {
                state.todolist.push({ todo: payload.todo, done: false });
            }
        },
        [Constant.DONE_TOGGLE]: (state, payload) => {
            state.todolist[payload.index].done = !state.todolist[payload.index].done;
        },
        [Constant.DELETE_TODO]: (state, payload) => {
            state.todolist.splice(payload.index, 1);
        }
    }
})

export default store;