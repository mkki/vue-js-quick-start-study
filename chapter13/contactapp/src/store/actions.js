import Constant from '../constant';
import ContactsAPI from '../api/ContactsAPI';

export default {
    [Constant.CHANGE_ISLOADING](store, payload) {
        store.commit(Constant.CHANGE_ISLOADING, payload);
    },
    async [Constant.ADD_CONTACT](store) {
        store.dispatch(Constant.CHANGE_ISLOADING, { isloading: true });
        var response = await ContactsAPI.addContact(store.state.contact);
        if (response.data.status == 'success') {
            store.dispatch(Constant.FETCH_CONTACTS, { pageno: 1 });
        } else {
            console.log(`연락처 추가 실패: ${response.data}`)
        }
    },
    async [Constant.UPDATE_CONTACT](store) {
        store.dispatch(Constant.CHANGE_ISLOADING, { isloading: true });
        var response = await ContactsAPI.updateContact(store.state.contact);
        var currentPageNo = store.state.contactlist.pageno;
        if (response.data.status == 'success') {
            store.dispatch(Constant.FETCH_CONTACTS, { pageno: currentPageNo });
        } else {
            console.log(`연락처 변경 실패: ${response.date}`);
        }
    },
    async [Constant.UPDATE_PHOTO](store, payload) {
        store.dispatch(Constant.CHANGE_ISLOADING, { isloading: true });
        var currentPageNo = store.state.contactlist.pageno;
        await ContactsAPI.updatePhoto(payload.no, payload.file);
        store.dispatch(Constant.FETCH_CONTACTS, { pageno: currentPageNo });
    },
    async [Constant.FETCH_CONTACTS](store, payload) {
        var pageno;
        if (typeof payload === 'undefined' || typeof payload.pageno === 'undefined') {
            pageno = 1;
        } else {
            pageno = payload.pageno;
        }
        var pagesize = store.state.contactlist.pagesize;

        store.dispatch(Constant.CHANGE_ISLOADING, { isloading: true });
        var response = await ContactsAPI.fetchContacts(pageno, pagesize);
        store.commit(Constant.FETCH_CONTACTS, { contactlist: response.data });
        store.dispatch(Constant.CHANGE_ISLOADING, { isloading: false });
    },
    async [Constant.DELETE_CONTACT](store, payload) {
        store.dispatch(Constant.CHANGE_ISLOADING, { isloading: true });
        var currentPageNo = store.state.contactlist.pageno;
        await ContactsAPI.deleteContact(payload.no);
        store.dispatch(Constant.FETCH_CONTACTS, { pageno: currentPageNo });
    },
    async [Constant.FETCH_CONTACT_ONE](store, payload) {
        store.dispatch(Constant.CHANGE_ISLOADING, { isloading: true });
        var response = await ContactsAPI.fetchContactOne(payload.no);
        store.commit(Constant.FETCH_CONTACT_ONE, { contact: response.data });
        store.dispatch(Constant.CHANGE_ISLOADING, { isloading: false });
    },
    [Constant.INITIALIZE_CONTACT_ONE](store) {
        store.commit(Constant.INITIALIZE_CONTACT_ONE);
    }
}