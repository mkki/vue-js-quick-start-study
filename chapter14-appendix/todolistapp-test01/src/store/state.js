export default {
    todolist: [
        { todo: "영화", done: false },
        { todo: "야구", done: false },
        { todo: "공부", done: false },
        { todo: "카페", done: false }
    ]
}