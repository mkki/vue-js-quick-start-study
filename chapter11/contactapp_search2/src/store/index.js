import Vue from 'vue';
import Vuex from 'vuex';
import Constant from '../Constant';
import SearchApi from '../api/SearchAPI';
import module1 from './module1';

Vue.use(Vuex);

const store = new Vuex.Store({
    state: {
        keywordlist: []
    },
    mutations: {
        [Constant.ADD_KEYWORD]: (state, payload) => {
            state.keywordlist.splice(0, 0, payload.name);
        }
    },
    actions: {
        [Constant.ADD_KEYWORD]: ({ state, commit }, payload) => {
            commit(Constant.ADD_KEYWORD, payload);
        }
    },
    modules: {
        m1: module1
    }
})

export default store;