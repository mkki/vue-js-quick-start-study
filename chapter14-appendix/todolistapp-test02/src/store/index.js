import Vue from 'vue';
import Vuex from 'vuex';
import Constant from '../constant';
import state from './state';
import mutations from './mutations'
import actions from './actions'
import es6promise from 'es6-promise'
es6promise.polyfill();

Vue.use(Vuex);

const store = new Vuex.Store({
    state,
    mutations,
    actions
})

export default store;