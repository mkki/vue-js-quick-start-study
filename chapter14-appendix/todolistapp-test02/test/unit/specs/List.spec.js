import Vue from 'vue'
import Vuex from 'vuex'
import store from '@/store'
import List from '@/components/List'
import constant from '@/constant'

describe('List.vue', () => {
    var vm;

    beforeAll(() => {
        vm = new Vue({
            store,
            render: h => h(List)
        }).$mount()

        vm.$store.dispatch(constant.ADD_TODO, { todo: 'jestTest' })
    })

    it('초기 렌더링 화면 테스트', () => {
        expect(vm.$el.querySelectorAll('li').length).toEqual(5);
    })

    it('스냅샷 테스트', (done) => {
        Vue.nextTick()
            .then(() => {
                expect(vm.$el).toMatchSnapshot();
                done();
            })
            .catch(done)
    })
})