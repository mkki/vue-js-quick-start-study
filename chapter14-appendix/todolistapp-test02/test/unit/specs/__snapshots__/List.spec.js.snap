// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List.vue 스냅샷 테스트 1`] = `
<ul>
  <li
    class=""
  >
    <span>
      영화
    </span>
     
    <!---->
     
    <span
      class="close"
    >
      ×
    </span>
  </li>
  <li
    class=""
  >
    <span>
      야구
    </span>
     
    <!---->
     
    <span
      class="close"
    >
      ×
    </span>
  </li>
  <li
    class=""
  >
    <span>
      공부
    </span>
     
    <!---->
     
    <span
      class="close"
    >
      ×
    </span>
  </li>
  <li
    class=""
  >
    <span>
      카페
    </span>
     
    <!---->
     
    <span
      class="close"
    >
      ×
    </span>
  </li>
  <li
    class=""
  >
    <span>
      jestTest
    </span>
     
    <!---->
     
    <span
      class="close"
    >
      ×
    </span>
  </li>
</ul>
`;
